### PR TITLE
Don't duplicate log failed futures in the http json api (commandservice specific ones)

### LINKED ISSUE #9985: JSON API logs command submission errors twice
Currently an error during a command submission will be logged twice which is a bit confusing and not all that useful:
```
10:32:38.715 [http-json-ledger-api-akka.actor.default-dispatcher-4] ERROR com.daml.http.CommandService - Symbol(createAndExercise) failure , context: {instance_uuid=2a5f9367-2d4b-496f-991e-550ac6f215aa, request_id=0be07af7-9c57-4c10-8ac2-c44ff2c4a82d}
io.grpc.StatusRuntimeException: INVALID_ARGUMENT: Command interpretation error in LF-DAMLe: Interpretation error: Error: User abort: Assertion failed. Details: Last location: [DA.Internal.Assert:19], partial transaction: root node NodeId(0): NodeCreate(ContractId(00e7b6f3fbf58a8fa7d782490e59c7f08de39c591c7ed612413af9921cef5d8fbb),4c549e43b26e1ce8822c89fbda2b7e9c7c2d81fa356a0294937d6df1b9b57096:Main:T,ValueRecord(Some(4c549e43b26e1ce8822c89fbda2b7e9c7c2d81fa356a0294937d6df1b9b57096:Main:T),ImmArray((Some(p),ValueParty(Alice)))),,Some(Location(d8c2ae4afe16f0192cfcaa00ee16e0cf68c22020aa81e1ad55f739f5c42516fd,DA.Internal.Prelude,$u002b$u002b,(372,25),(372,27))),TreeSet(Alice),TreeSet(Alice),None,V11).
        at io.grpc.Status.asRuntimeException(Status.java:534)
        at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:533)
        at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:553)
        at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:68)
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:739)
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:718)
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
10:32:38.715 [http-json-ledger-api-akka.actor.default-dispatcher-4] ERROR com.daml.http.Endpoints - Future failed , context: {instance_uuid=2a5f9367-2d4b-496f-991e-550ac6f215aa, request_id=0be07af7-9c57-4c10-8ac2-c44ff2c4a82d}
io.grpc.StatusRuntimeException: INVALID_ARGUMENT: Command interpretation error in LF-DAMLe: Interpretation error: Error: User abort: Assertion failed. Details: Last location: [DA.Internal.Assert:19], partial transaction: root node NodeId(0): NodeCreate(ContractId(00e7b6f3fbf58a8fa7d782490e59c7f08de39c591c7ed612413af9921cef5d8fbb),4c549e43b26e1ce8822c89fbda2b7e9c7c2d81fa356a0294937d6df1b9b57096:Main:T,ValueRecord(Some(4c549e43b26e1ce8822c89fbda2b7e9c7c2d81fa356a0294937d6df1b9b57096:Main:T),ImmArray((Some(p),ValueParty(Alice)))),,Some(Location(d8c2ae4afe16f0192cfcaa00ee16e0cf68c22020aa81e1ad55f739f5c42516fd,DA.Internal.Prelude,$u002b$u002b,(372,25),(372,27))),TreeSet(Alice),TreeSet(Alice),None,V11).
        at io.grpc.Status.asRuntimeException(Status.java:534)
        at io.grpc.stub.ClientCalls$UnaryStreamToFuture.onClose(ClientCalls.java:533)
        at io.grpc.internal.ClientCallImpl.closeObserver(ClientCallImpl.java:553)
        at io.grpc.internal.ClientCallImpl.access$300(ClientCallImpl.java:68)
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInternal(ClientCallImpl.java:739)
        at io.grpc.internal.ClientCallImpl$ClientStreamListenerImpl$1StreamClosed.runInContext(ClientCallImpl.java:718)
        at io.grpc.internal.ContextRunnable.run(ContextRunnable.java:37)
        at io.grpc.internal.SerializingExecutor.run(SerializingExecutor.java:123)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

### PR DESCRIPTION
**Pull request by realvictorprm**
fixes #9985

This removes duplicate logging for me. Tested this manually with main and with my branch and the duplicate did not appear anymore with my branch (just send a command req with a wrong party).

changelog_begin

- [JSON API] Errors which arise from the CommandService are not logged twice anymore (thus reducing noise)

changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by cocreature — APPROVED** [review]
LGTM but I’m a bit worried this gets out of sync and we start dropping logs.

Maybe a more reliable way of handling this would be to turn the error in the command service into a `Left` and then we track at the types if we still have to log it:
1. Future failures are logged
2. Left has already been handled and logged.

**Comment by realvictorprm** [comment]
I share your worries here, however I didn't come up with a better solution back then yesterday. I will try to apply the suggestion you made.

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/CommandService.scala [hunk 1]
```diff
@@ -248,11 +256,16 @@ class CommandService(
 }
 
 object CommandService {
-  final case class Error(id: Symbol, message: String)
+  sealed trait Error
+  final case class FutureFailedButAlreadyLogged(err: Throwable) extends Error
+  final case class CommandError(id: Symbol, message: String) extends Error
```
Comment (cocreature) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/CommandService.scala:261]: I don’t think the separate types help here. Can we just have a single type and if needed make the id optional?

Comment (realvictorprm) [reply ledger-service/http-json/src/main/scala/com/digitalasset/http/CommandService.scala:262]: Well that would be an option too I guess :thinking:

Comment (realvictorprm) [reply ledger-service/http-json/src/main/scala/com/digitalasset/http/CommandService.scala]: adressed with 9fcdba8

**Review by cocreature — APPROVED** [review]
Thanks!

**Comment by realvictorprm** [comment]
Anytime :smiley:

**Pull request merged**